interfaces/final-object.interface.ts

```diff
@@ -1,19 +1,27 @@
-import { ResolutionString } from '../src/app/pipes/resolution-filter.service';
-
 export type StarRating = 0.5 | 1.5 | 2.5 | 3.5 | 4.5 | 5.5;
 
 // must be heights from true `16:9` resolutions AND divisible by 8
 export type AllowedScreenshotHeight = 144 | 216 | 288 | 360 | 432 | 504;
 
+export type ResolutionString = '' | 'SD' | '720' | '720+' | '1080' | '1080+' | '4K' | '4K+';
+
+export interface SourceFolder {
+  path: string;
+  watch: boolean;
+}
+
+export type InputSources = Record<number, SourceFolder>;
+
 export interface FinalObject {
-  addTags?: string[];           // tags to add
-  hubName: string;              // the name of the hub -- for recently-opened
-  images: ImageElement[];       // see below
-  inputDir: string;             // later may support array of many input directories
-  numOfFolders: number;         // number of folders
-  removeTags?: string[];        // tags to remove
+  addTags: string[];             // tags to add
+  hubName: string;               // the name of the hub -- for recently-opened
+  images: ImageElement[];
+  // inputDir: string;           // became `inputDirs` in VHA3 - gets upgraded to `inputDirs` on first open in VHA3
```

Comment:
Remove comment in favour of adding it to the `upgrade` class in my other comment?